tarok/src/state.cpp

```diff
@@ -145,10 +145,18 @@ std::vector<open_spiel::Action> TarokState::LegalActionsInTalonExchange()
     std::iota(actions.begin(), actions.end(), 0);
     return actions;
   }
-  // discarding the cards
+  // prevent exchange of taroks and kings
   std::vector<open_spiel::Action> actions;
   for (auto const& action : players_cards_.at(current_player_)) {
-    if (ActionToCard(action).points != 5) actions.push_back(action);
+    auto card = ActionToCard(action);
```

Comment:
This should be `const Card& card = ...` so that we actually avoid copying the card (even if the method returns a reference and the caller assigns it to a value, it is copied).

Reply:
Replaced with const auto& card = ...